tools/payments/cmd/authorize/main.go

```diff
@@ -109,35 +110,73 @@ func main() {
 			}
 			defer f.Close()
 
-			var report payments.AttestedReport
-			if err := payments.ReadReport(&report, f); err != nil {
+			var report paymentscli.AttestedReport
+			if err := paymentscli.ReadReportFromResponses(&report, f); err != nil {
 				log.Fatalf("failed to read report from stdin: %v\n", err)
 			}
 
-			wc.Count += len(report)
+			if report[0].PayoutID != *payoutID {
+				log.Fatalf("payoutID did not match report: %s\n", report[0].PayoutID)
+			}
+
+			preparedReportFile, err := os.Open(*preparedReportFilename)
+			if err != nil {
+				log.Fatalf("failed to open prepared report file: %v\n", err)
+			}
+			defer preparedReportFile.Close()
```

Comment:
This is not going to work because of the `Fatal` below (and above). Don't call `Fatal` from random places. Log with error and return, if returning an error is not an option.